specs/CriticalRestartRequired.md

```diff
@@ -0,0 +1,105 @@
+Critical Update Available
+===
+
+# Background
+As WebView2 developers, we often have to author ECS (experimentation and configuration service - 
+used to perform A/B testing or remotely disable features) configurations to toggle feature flags. 
+However, once these payloads are received, there is no way to restart the WebView2 and apply the
+payload. The purpose of this API is to detect such critical payloads and inform the end developer
+so they may restart their app or their WebView2 or other appropriate action.
+
+# Examples
+## WinRT and .NET   
+```c#
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    if (e.IsSuccess)
+    {
+        // ...
+        webView.CoreWebView2.Environment.CriticalRestartRequired += WebView_CriticalRestartRequired;
+    }
+}
+
+void WebView_CriticalRestartRequired(object sender, object e)
+{
+    // Depending on your app experience, you may or may not
+    // want to prompt user to restart the app.
+    RestartIfSelectedByUser();
+}
+```
+
+## Win32 C++
+```cpp
+void CoreWebView2InitializationCompleted() {
+    wil::com_ptr<ICoreWebView2Environment> m_webViewEnvironment;
```

Comment:
Declare this as a member variable rather than a local, and then the reader realizes that it's initialized somewhere out-of-scope to this example

Reply:
Please fix.